src/psyclone/psyGen.py

```diff
@@ -2704,6 +2704,8 @@ def zero_reduction_variable(self, parent, position=None):
 
         '''
         from psyclone.f2pygen import AssignGen, DeclGen, AllocateGen
+        from psyclone.domain.lfric import LFRicConstants
```

Comment:
I think that
- `Argument` class should have abstract property for kind that can then be overriden in children classes for each API,
- `DynKernelArgument` can then specify the mapping and use the information from parsing Algorithms where mathcing the type of data and kind is ambivalent (such as fields),
- This method, `zero_reduction_variable` should really live in `DynKern` as it is only applicable for the LFRic API (perhaps an abstract method should be kept here).

What do you think? It seems as a separate PR to me!

Reply:
Yes, I agree to the first two points - that is essentially what I just said in the previous comment so it sounds like we agree :-). As for the `zero_reduction_variable` issue, I don't think I agree. What it is doing is generic for a reduction (zero-ing the values) and is only part of the reduction, there are other generic methods that call this one. So we should either support reductions in all APIs (even if we perhaps do not use them) or not. As the problem is really just about getting API-specific precision, I think the addition of a kind or precision method in the Argument class that returns the required information (which could be None) would make this method generic. I agree that this is probably a separate PR, but would need to be done before this one!

Reply:
I introduced the `precision` property in the `KernelArgument` class and initialised it to `None`. I also adjusted the `zero_reduction_variable` function to set the zero sum variable to a zero value only if the precision is not provided. Otherwise it sets it to a zero value plus precision. I added the appropriate test where I hacked the precision of `DATA_TYPE_MAP["scalar"]` to be an empty string to test the zero summation variable with no precision information. I also used the opportunity to tidy up reduction tests (they did not really employ distributed memory).